File: java/client/test/com/thoughtworks/selenium/corebased/TestGettingValueOfCheckbox.java
package com.thoughtworks.selenium.corebased;

import com.thoughtworks.selenium.InternalSelenseTestBase;
import org.junit.Test;

public class TestGettingValueOfCheckbox extends InternalSelenseTestBase {
	@Test public void testGettingValueOfCheckbox() throws Exception {
    selenium.open("../tests/html/test_submit.html");

    String elementLocator = "name=okayToSubmit";
    assertEquals("off", selenium.getValue(elementLocator));

    selenium.click(elementLocator);
    assertEquals("on", selenium.getValue(elementLocator));

    selenium.click(elementLocator);
    assertEquals("off", selenium.getValue(elementLocator));

	}
}

File: dotnet/test/Selenium.WebDriverBackedSelenium.Tests/TestGettingValueOfCheckbox.cs
using System;
using System.Collections.Generic;
using System.Text;
using NUnit.Framework;
using System.Text.RegularExpressions;
using System.Threading;

namespace Selenium.Tests
{
    [TestFixture]
    public class TestGettingValueOfCheckbox : SeleniumTestCaseBase
    {
        [Test]
        public void ShouldBeAbleToHandleFunkyEventHandling()
        {
            selenium.Open("../tests/html/test_submit.html");

            string elementLocator = "name=okayToSubmit";
            Assert.AreEqual("off", selenium.GetValue(elementLocator));

            selenium.Click(elementLocator);
            Assert.AreEqual("on", selenium.GetValue(elementLocator));

            selenium.Click(elementLocator);
            Assert.AreEqual("off", selenium.GetValue(elementLocator));
        }
    }
}
